src/VisualStudio/CSharp/Impl/ProjectSystemShim/TempPECompilerService.cs

```diff
@@ -59,7 +59,9 @@ public void CompileTempPE(string pszOutputFileName, int sourceCount, string[] fi
                     .WithXmlReferenceResolver(XmlFileResolver.Default)
                     .WithMetadataReferenceResolver(metadataResolver));
 
-            compilation.Emit(pszOutputFileName);
+            var result = compilation.Emit(pszOutputFileName);
+
+            Contract.ThrowIfFalse(result.Success);
```

Comment:
Is this the right way to handle TempPE build failures? or should we be writing to output window & error list window?